src/TinyUPnP.cpp

```diff
@@ -678,6 +685,9 @@ ssdpDeviceNode* TinyUPnP::listSsdpDevices() {
 // Note: the response from the IGD is sent back as unicast to this device
 // Note: only gateway defined IGD response will be considered, the rest will be ignored
 ssdpDevice* TinyUPnP::waitForUnicastResponseToMSearch(IPAddress gatewayIP) {
+    // Flush the UDP buffer since otherwise anyone who responded first will be the only one we see
+    // and we will not see the response from the gateway router
+    _udpClient.flush();
```

Comment:
do you mean that this will happen in cases where we exit the method before reading from `_udpClient` (e.g. when the gateway IP is different from the remote IP in the packet)? In this case, the next time we enter the method we will be in an infinite loop?